### PR TITLE
Search for alerts by coordinates

### PR DESCRIPTION
**Pull request by acrellin**
This PR allows users to search for alerts by location.

alerts_by_coords_table_3

Closes https://github.com/fritz-marshal/fritz-beta-feedback/issues/75

### CODE REVIEW

**Comment by acrellin** [comment]
Forgot the ra/dec columns, will update

File: fritz.defaults.yaml
Comment (dannygoldstein) [inline fritz.defaults.yaml:80]: ?

File: extensions/skyportal/skyportal/app_server_fritz.py
Comment (dannygoldstein) [inline extensions/skyportal/skyportal/app_server_fritz.py:21]: is `(.*)` needed?

File: extensions/skyportal/skyportal/handlers/api/alert.py
Comment (dannygoldstein) [inline extensions/skyportal/skyportal/handlers/api/alert.py:1090]: can we add descriptions of the parameters here. specifically, what are the units for radius (degrees or arcsec)?

File: extensions/skyportal/skyportal/handlers/api/alert.py
Comment (dannygoldstein) [inline extensions/skyportal/skyportal/handlers/api/alert.py:1124]: possible to use a marshmallow schema here?

File: extensions/skyportal/skyportal/handlers/api/alert.py
Comment (dannygoldstein) [inline extensions/skyportal/skyportal/handlers/api/alert.py:1125]: It seems like `get_user_streams()` does the same thing as `self.associated_user_object.streams`:

```python
class ZTFAlertHandler(BaseHandler):
    def get_user_streams(self):

        streams = (
            DBSession()
            .query(Stream)
            .join(StreamUser)
            .filter(StreamUser.user_id == self.associated_user_object.id)
            .all()
        )
        if streams is None:
            streams = []

        return streams
```

Is there an advantage to using this over simply `self.associated_user_object.streams`?

File: extensions/skyportal/static/js/components/ZTFAlertsByCoords.jsx
Comment (dannygoldstein) [inline extensions/skyportal/static/js/components/ZTFAlertsByCoords.jsx:232]: Can we also add ML score?

File: fritz.defaults.yaml
Comment (acrellin) [reply fritz.defaults.yaml:80]: This allows this to work on Linux

Comment (dannygoldstein) [reply fritz.defaults.yaml:80]: Where does 172.17.0.1 come from?

File: extensions/skyportal/skyportal/app_server_fritz.py
Comment (acrellin) [reply extensions/skyportal/skyportal/app_server_fritz.py:21]: No. Removed

File: extensions/skyportal/skyportal/handlers/api/alert.py
Comment (acrellin) [reply extensions/skyportal/skyportal/handlers/api/alert.py:1090]: Done

File: extensions/skyportal/skyportal/handlers/api/alert.py
Comment (acrellin) [reply extensions/skyportal/skyportal/handlers/api/alert.py:1124]: Possible, but I'd rather not.

File: extensions/skyportal/skyportal/handlers/api/alert.py
Comment (acrellin) [reply extensions/skyportal/skyportal/handlers/api/alert.py:1125]: Ask @dmitryduev -- I'm just mirroring what he's already implemented in other handlers here (the function is not new to this PR)

Comment (acrellin) [reply extensions/skyportal/skyportal/handlers/api/alert.py:1125]: Resolving as out of scope

File: extensions/skyportal/static/js/components/ZTFAlertsByCoords.jsx
Comment (acrellin) [reply extensions/skyportal/static/js/components/ZTFAlertsByCoords.jsx:232]: What field is that? I don't see anything obvious here https://docs.fritz.science/catalogs.html#ztf-alerts

**Comment by dannygoldstein** [comment]
Also, can we add integration tests ?

File: fritz.defaults.yaml
Comment (acrellin) [reply fritz.defaults.yaml:80]: @dmitryduev

File: extensions/skyportal/static/js/components/ZTFAlertsByCoords.jsx
Comment (dannygoldstein) [reply extensions/skyportal/static/js/components/ZTFAlertsByCoords.jsx:232]: drb

Comment (acrellin) [reply extensions/skyportal/static/js/components/ZTFAlertsByCoords.jsx:232]: is rb relevant too?

Comment (acrellin) [reply extensions/skyportal/static/js/components/ZTFAlertsByCoords.jsx:232]: and where are these fields documented?

Comment (dannygoldstein) [reply extensions/skyportal/static/js/components/ZTFAlertsByCoords.jsx:232]: rb is not relevant. just drb. These fields are documented here 

https://zwickytransientfacility.github.io/ztf-avro-alert/schema.html

**Comment by acrellin** [comment]
@dannygoldstein tests in, comments addressed

File: extensions/skyportal/static/js/components/ZTFAlertsByCoords.jsx
Comment (acrellin) [reply extensions/skyportal/static/js/components/ZTFAlertsByCoords.jsx:234]: then why is `rb` shown here along with `drb`? https://github.com/fritz-marshal/fritz/blob/master/extensions/skyportal/static/js/components/ZTFAlert.jsx#L221

Comment (acrellin) [reply extensions/skyportal/static/js/components/ZTFAlertsByCoords.jsx:234]: I've added both, as that's what Dima did on the above-linked alerts page

Comment (dannygoldstein) [reply extensions/skyportal/static/js/components/ZTFAlertsByCoords.jsx:234]: rb is an older model that performs abysmally compared to drb and is no longer used in production

**Comment by dannygoldstein** [comment]
can we remove rb please? still waiting on response re: ip address above.

**Comment by acrellin** [comment]
> can we remove rb please? still waiting on response re: ip address above.

@dannygoldstein Why is it relevant in ZTFAlerts.jsx but not here?

**Comment by dannygoldstein** [comment]
i see it is there, but I don't think it is relevant there either. @dmitryduev does anyone actually use rb?

**Comment by acrellin** [comment]
@dannygoldstein removed changes to fritz.yaml.defaults and the rb column

**Comment by acrellin** [comment]
Latest table appearance/behavior:
alerts_by_coords_table_2

**Comment by acrellin** [comment]
@dannygoldstein @dmitryduev ready for review

**Review by dannygoldstein — APPROVED** [review]

File: fritz.defaults.yaml
Comment (dannygoldstein) [reply fritz.defaults.yaml]: i'm assuming this will get overwritten in the upcoming fritz tool refactor so approving with that expectation in mind.

Comment (acrellin) [reply fritz.defaults.yaml]: This is no longer in the diff - change has been removed

File: extensions/skyportal/static/js/components/ZTFAlertsByCoords.jsx
Comment (dmitryduev) [inline extensions/skyportal/static/js/components/ZTFAlertsByCoords.jsx:146]: Let's also render the acai scores:

```suggestion
      drb: alert?.candidate.drb,
      acai_h: alert?.classifications?.acai_h,
      acai_n: alert?.classifications?.acai_n,
      acai_o: alert?.classifications?.acai_o,
      acai_v: alert?.classifications?.acai_v,
      acai_b: alert?.classifications?.acai_b,
```

File: extensions/skyportal/static/js/components/ZTFAlertsByCoords.jsx
Comment (dmitryduev) [inline extensions/skyportal/static/js/components/ZTFAlertsByCoords.jsx:281]: Please add the acai scores (see above).

File: extensions/skyportal/static/js/components/ZTFAlertsByCoords.jsx
Comment (dmitryduev) [inline extensions/skyportal/static/js/components/ZTFAlertsByCoords.jsx:333]: Who said it'll work next time lol? :)

```suggestion
    return <div>Failed to fetch alert data.</div>;
```

File: extensions/skyportal/static/js/ducks/alertsByCoords.js
Comment (dmitryduev) [inline extensions/skyportal/static/js/ducks/alertsByCoords.js:11]: Add units here that default to arcsec?

File: fritz.defaults.yaml
Comment (dmitryduev) [inline fritz.defaults.yaml:65]: I suggest we do `/alerts_by_coords/ztf` since what we have here is very ZTF-specific, and we may/will implement this for other surveys (e.g. PGIR). + use query parameters as discussed above. @dannygoldstein you agree?

File: extensions/skyportal/static/js/components/ZTFAlertsByCoords.jsx
Comment (dmitryduev) [inline extensions/skyportal/static/js/components/ZTFAlertsByCoords.jsx:352]: Please expand the table to the full Paper width.

File: extensions/skyportal/skyportal/handlers/api/alert.py
Comment (dmitryduev) [inline extensions/skyportal/skyportal/handlers/api/alert.py:1163]: Let's put in some noob protection (return at most a thousand objects):

```suggestion
            },
            "kwargs": {
                "limit": 1000
            }
```

File: extensions/skyportal/skyportal/handlers/api/alert.py
Comment (dmitryduev) [reply extensions/skyportal/skyportal/handlers/api/alert.py:1093]: And add radius_units? @dannygoldstein what do you think?

File: extensions/skyportal/skyportal/handlers/api/alert.py
Comment (dmitryduev) [inline extensions/skyportal/skyportal/handlers/api/alert.py:1115]: If we end up making radius units configurable (@dannygoldstein should we?), add some checks as in https://github.com/dmitryduev/kowalski/blob/master/kowalski/api.py#L736.

File: extensions/skyportal/skyportal/handlers/api/alert.py
Comment (dmitryduev) [inline extensions/skyportal/skyportal/handlers/api/alert.py:1125]: Let's put in a check that the radius is not more than, say 1 degree - noob protection.

File: extensions/skyportal/skyportal/handlers/api/alert.py
Comment (dmitryduev) [inline extensions/skyportal/skyportal/handlers/api/alert.py:132]: Let's change the default timeout to 10 seconds on line 125.

**Review by dmitryduev — CHANGES_REQUESTED** [review]
Looks reaaaally nice @acrellin, great work! A few minor-ish requests.

File: extensions/skyportal/static/js/components/ZTFAlertsByCoords.jsx
Comment (dmitryduev) [inline extensions/skyportal/static/js/components/ZTFAlertsByCoords.jsx:242]: One more request @acrellin: could you make the objectId's clickable (or turn into obvious buttons?) and point them at `/alerts/ztf/<objectId>`?

File: extensions/skyportal/static/js/components/Alerts.jsx
Comment (acrellin) [reply extensions/skyportal/static/js/components/Alerts.jsx:87]: @dmitryduev this isn't the API endpoint, just the react-router route

Comment (acrellin) [reply extensions/skyportal/static/js/components/Alerts.jsx:87]: When the API call is made, it indeed uses query params, e.g. `"?ra=..."`

File: extensions/skyportal/static/js/components/ZTFAlertsByCoords.jsx
Comment (acrellin) [reply extensions/skyportal/static/js/components/ZTFAlertsByCoords.jsx:149]: okay, I just copied what was in the ZTFAlerts page

**Comment by acrellin** [comment]
Thanks for the review @dmitryduev . Will update accordingly.

File: extensions/skyportal/static/js/components/ZTFAlertsByCoords.jsx
Comment (dmitryduev) [reply extensions/skyportal/static/js/components/ZTFAlertsByCoords.jsx:149]: oh, oops, could you fix it there as well, please? :)

File: extensions/skyportal/static/js/components/Alerts.jsx
Comment (dmitryduev) [reply extensions/skyportal/static/js/components/Alerts.jsx:87]: Right... found this: https://learnwithparam.com/blog/how-to-handle-query-params-in-react-router/
As said, I like it the way you implemented it, it's just that it's prob something that people are more familiar with. @dannygoldstein @profjsb what would you guys say?

File: extensions/skyportal/static/js/components/ZTFAlertsByCoords.jsx
Comment (acrellin) [reply extensions/skyportal/static/js/components/ZTFAlertsByCoords.jsx]: Done: 
alerts_by_coords_table_button_objid

File: extensions/skyportal/skyportal/handlers/api/alert.py
Comment (acrellin) [reply extensions/skyportal/skyportal/handlers/api/alert.py:1139]: Should we still ensure radius is no more than one degree if we're limiting the number of results in the query?

**Comment by acrellin** [comment]
@dmitryduev feedback addressed. New table screenshot:



Ready for re-review!

File: extensions/skyportal/skyportal/handlers/api/alert.py
Comment (dmitryduev) [inline extensions/skyportal/skyportal/handlers/api/alert.py:1180]: Thinking about this, since we're essentially limiting the alert contents to the size of ~20 floats, I think we can safely bump the limit up to 10000.
```suggestion
                "limit": 10000,
```

File: fritz.defaults.yaml
Comment (dmitryduev) [reply fritz.defaults.yaml:65]: I still think we should do /alerts_by_coords/ztf/...

**Review by dmitryduev — APPROVED** [review]
Looks great, thanks @acrellin! A few minor comments/requests, please feel free to self-merge once implemented.

File: extensions/skyportal/skyportal/handlers/api/alert.py
Comment (dmitryduev) [reply extensions/skyportal/skyportal/handlers/api/alert.py:1180]: @acrellin Thinking about this a bit more, could you please add a tooltip to the search box saying that the max radius is 1 deg and the results will be limited to 10,000 nearest matches?

**Comment by acrellin** [comment]
Updated:

alerts_by_coords_table_3

**Comment by acrellin** [comment]
Passing locally. 
```
======================= 4 passed, 572 warnings in 0.65s ======================== 
```
All API tests are currently failing on CI. Will fix in an upcoming PR.

**Pull request merged**
